projects/templarfi/index.js

```diff
@@ -0,0 +1,403 @@
+    ])
+
+    /** @type {Snapshot} */
+    const snapshot = snapshotRaw
+    /** @type {Configuration} */
+    const configuration = configurationRaw
+
+    validateSnapshot(snapshot)
+    validateConfiguration(configuration)
+
+    if (!configuration?.borrow_asset || !configuration?.collateral_asset) {
+        throw new Error('Invalid configuration: missing borrow_asset or collateral_asset')
+    }
+
+    const borrowAssetToken = extractTokenAddress(configuration.borrow_asset, 'borrow')
+    const collateralAssetToken = extractTokenAddress(configuration.collateral_asset, 'collateral')
+
+    // Calculate net liquidity in raw amounts
+    const totalBorrowDeposited = BigNumber(snapshot.borrow_asset_deposited_active)
+    const totalBorrowIncoming = BigNumber(snapshot.borrow_asset_deposited_incoming)
+    const netCollateral = BigNumber(snapshot.collateral_asset_deposited)
+    const totalBorrowed = BigNumber(snapshot.borrow_asset_borrowed)
+    const netBorrowed = totalBorrowDeposited.plus(totalBorrowIncoming).minus(totalBorrowed)
+
+    return { borrowAssetToken, collateralAssetToken, netBorrowed, netCollateral }
+}
+
+/**
+ * Calculate Total Value Locked (TVL) for Templar Protocol
+ *
+ * OVERALL TVL CALCULATION:
+ * 1. For each market deployment:
+ *    - Calculate net borrow asset liquidity = total borrow deposits - outstanding loans
+ *    - Include full collateral deposits
+ * 2. Return all borrow and collateral assets with their net amounts
+ *
+ * This methodology follows DeFi standards where TVL represents actual value
+ * locked in the protocol, accounting for outstanding obligations.
+ */
+async function tvl() {
+    const balances = {}
+
+    try {
+        const deployments = await fetchAllDeployments(TEMPLAR_REGISTRY_CONTRACTS)
+
+        if (deployments.length === 0) {
+            console.log('No Templar deployments found')
+            return balances
+        }
+
+        const results = await Promise.allSettled(
+            deployments.map(processMarket)
+        )
+
+        // Process results and add to balances
+        results.forEach((result, index) => {
+            if (result.status === 'fulfilled') {
+                const { borrowAssetToken, collateralAssetToken, netBorrowed, netCollateral } = result.value
+
+                sumSingleBalance(balances, borrowAssetToken, netBorrowed.toFixed())
+                sumSingleBalance(balances, collateralAssetToken, netCollateral.toFixed())
+            } else {
+                console.log(`Error processing market ${deployments[index]}:`, result.reason)
+            }
+        })
+
+    } catch (err) {
+        console.log(`Error fetching deployments:`, err)
+    }
+
+    return balances
+}
+
+module.exports = {
+    methodology: 'TVL is calculated by summing the net borrow asset liquidity (deposits minus outstanding loans) and full collateral deposits for each market deployment.',
+    start: 1754902109, // August 11, 2025 08:48:29+UTC
+    near: {
+        tvl,
```

Comment:
please also export a `borrowed` property with the sum of all borrowed funds. `tvl` should export available liquidity (not including borrows)